### ISSUE TITLE
1048, Column 'logoff_date' cannot be null

### ISSUE THREAD

**Issue by paulipa** [pvyParts/allianceauth-corpstats-two#8]
Trying to update corporation details for our largest corporation with 10k+ members i get this following error:
```
[21/Jul/2020 06:37:50] ERROR [django.request:228] Internal Server Error: /corpstat/98258093/update/
Traceback (most recent call last):
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/db/backends/mysql/base.py", line 71, in execute
    return self.cursor.execute(query, args)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/MySQLdb/cursors.py", line 206, in execute
    res = self._query(query)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/MySQLdb/cursors.py", line 319, in _query
    db.query(q)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/MySQLdb/connections.py", line 259, in query
    _mysql.connection.query(self, query)
MySQLdb._exceptions.OperationalError: (1048, "Column 'logoff_date' cannot be null")

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/allianceauth/authentication/decorators.py", line 35, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/corpstats/views.py", line 46, in check_corpstats
    return view(request, corpstats, corp_id=corp_id)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/corpstats/views.py", line 139, in corpstats_update
    corpstats.update()
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/corpstats/models.py", line 101, in update
    CorpMember.objects.bulk_create(member_db_create)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/db/models/query.py", line 474, in bulk_create
    ids = self._batched_insert(objs_without_pk, fields, batch_size, ignore_conflicts=ignore_conflicts)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/db/models/query.py", line 1211, in _batched_insert
    self._insert(item, fields=fields, using=self.db, ignore_conflicts=ignore_conflicts)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/db/models/query.py", line 1186, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 1375, in execute_sql
    cursor.execute(sql, params)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/db/backends/utils.py", line 99, in execute
    return super().execute(sql, params)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/home/allianceserver/venv/auth/lib/python3.6/site-packages/django/db/backends/mysql/base.py", line 76, in execute
    raise utils.IntegrityError(*tuple(e.args))
django.db.utils.IntegrityError: (1048, "Column 'logoff_date' cannot be null")
```
Could it be possible that we have members in the corporation who have just started playing and have not logged off for the first time yet? Would allowing this column value to be null cause any issues somewhere?

This issue does not seem to happen with our smaller corporations

**Comment by pvyParts** [comment]
Yep, I thought I caught all of these I'll have another look this evening.

**Comment by pvyParts** [comment]
this should be fixed from `v1.1.2`

can you update and confirm and or report back if it is happening on the latest version.

**Comment by paulipa** [comment]
seems to be working thanks!
